packages/components/src/components/contract-card/contract-card-items/contract-card-dialog.jsx

```diff
@@ -4,63 +4,53 @@ import ReactDOM from 'react-dom';
 import { CSSTransition } from 'react-transition-group';
 import './sass/contract-card-dialog.scss';
 
-class ContractCardDialog extends React.Component {
-    constructor(props) {
-        super(props);
-        this.ref = React.createRef();
-    }
+const ContractCardDialog = React.forwardRef(({ children, is_visible, left, toggleDialog, toggle_ref, top }, ref) => {
+    React.useEffect(() => {
+        document.addEventListener('mousedown', handleClickOutside, true);
+        return () => document.removeEventListener('mousedown', handleClickOutside, true);
+    });
 
-    componentDidMount() {
-        document.addEventListener('mousedown', this.handleClickOutside, true);
-    }
-
-    componentWillUnmount() {
-        document.removeEventListener('mousedown', this.handleClickOutside, true);
-    }
-
-    handleClickOutside = event => {
-        if (this.ref && this.ref.current && this.props.is_visible) {
-            if (this.ref.current.contains(event.target)) {
+    const handleClickOutside = event => {
```

Comment:
is it possible to reuse [useOnClickoutside](packages/components/src/hooks/use-onclickoutside.js) hook?

Reply:
Also this will run on every update. I think we need to run this just on `mount/unMount`.

Reply:
Yes, this is done. Thank you 👍